pkg/controller/dataimportcron-controller.go

```diff
@@ -0,0 +1,532 @@
+		log.Error(err, "error listing dvs")
+		return err
+	}
+	recentImports := len(dvList.Items)
+	log.Info("recentImports", "count", recentImports)
+	if recentImports <= recentImportsToKeepPerCronJob {
+		return nil
+	}
+	sort.Slice(dvList.Items, func(i, j int) bool {
+		return dvList.Items[i].CreationTimestamp.Time.Before(dvList.Items[j].CreationTimestamp.Time)
+	})
+	for _, dv := range dvList.Items {
+		if recentImports <= recentImportsToKeepPerCronJob {
+			break
+		}
+		if err := r.client.Delete(context.TODO(), &dv); err != nil {
+			if k8serrors.IsNotFound(err) {
+				log.Info("DataVolume not found for deletion", "name", dv.Name)
+				continue
+			}
+			log.Error(err, "Unable to delete DataVolume", "name", dv.Name)
+			return err
+		}
+		log.Info("DataVolume deleted", "name", dv.Name)
+		recentImports--
+	}
+	return nil
+}
+
+// NewDataImportCronController creates a new instance of the DataImportCron controller
+func NewDataImportCronController(mgr manager.Manager, log logr.Logger) (controller.Controller, error) {
+	uncachedClient, err := client.New(mgr.GetConfig(), client.Options{
+		Scheme: mgr.GetScheme(),
+		Mapper: mgr.GetRESTMapper(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	reconciler := &DataImportCronReconciler{
+		client:         mgr.GetClient(),
+		uncachedClient: uncachedClient,
+		recorder:       mgr.GetEventRecorderFor(dataImportControllerName),
+		scheme:         mgr.GetScheme(),
+		log:            log.WithName(dataImportControllerName),
+	}
+	dataImportCronController, err := controller.New(dataImportControllerName, mgr, controller.Options{Reconciler: reconciler})
+	if err != nil {
+		return nil, err
+	}
+	if err := addDataImportCronControllerWatches(mgr, dataImportCronController, log); err != nil {
+		return nil, err
+	}
+	log.Info("Initialized DataImportCron controller")
+	return dataImportCronController, nil
+}
+
+func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Controller, log logr.Logger) error {
+	if err := cdiv1.AddToScheme(mgr.GetScheme()); err != nil {
+		return err
+	}
+	if err := imagev1.AddToScheme(mgr.GetScheme()); err != nil {
+		return err
+	}
+	if err := c.Watch(&source.Kind{Type: &cdiv1.DataImportCron{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return err
+	}
+	if err := c.Watch(&source.Kind{Type: &cdiv1.DataVolume{}}, &handler.EnqueueRequestForOwner{
+		OwnerType:    &cdiv1.DataImportCron{},
+		IsController: true,
+	}); err != nil {
+		return err
+	}
+	if err := c.Watch(&source.Kind{Type: &v1beta1.CronJob{}}, &handler.EnqueueRequestForOwner{
+		OwnerType:    &cdiv1.DataImportCron{},
+		IsController: true,
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func newCronJob(cron *cdiv1.DataImportCron) *v1beta1.CronJob {
+	var container corev1.Container
+	if cron.Spec.Source.Registry.URL != nil {
+		container = corev1.Container{
+			Name:  "skopeo",
+			Image: "quay.io/skopeo/stable",
```

Comment:
Can't hard code images like this. People might want to use a different image for this.

Reply:
Sure